vars/buildPlugin.groovy

```diff
@@ -106,11 +107,18 @@ def call(Map params = [:]) {
                                     mavenOptions += "-DskipTests"
                                 }
                                 mavenOptions += "clean install"
+                                if (!skipTests && first && enableCoverage) {
+                                    mavenOptions += "jacoco:prepare-agent test jacoco:report"
+                                }
                                 try {
+                                    echo "Running maven with options " + mavenOptions
                                     infra.runMaven(mavenOptions, jdk, null, null, addToolEnv)
                                 } finally {
                                     if (!skipTests) {
                                         junit('**/target/surefire-reports/**/*.xml,**/target/failsafe-reports/**/*.xml,**/target/invoker-reports/**/*.xml')
+                                        if (first && enableCoverage) {
+                                            publishCoverage adapters: [jacocoAdapter('target/site/jacoco/jacoco.xml')]
```

Comment:
should we enable the `calculateDiffForChangeRequests: true` ?

https://www.jenkins.io/blog/2019/07/21/Jenkins-code-coverage-diff-for-pull-request/#how-to-enable-code-coverage-diff-for-pull-requests

Reply:
what happens if this is a multi module project?

does the path need to be configurable or should we locate all the jacoco reports?

or YAGNI and enhance later?

Reply:
Yes, adding the diff would make sense, yes. I never used that before but it sounds quite useful.